component/client/ClientCore.tsx

```diff
@@ -1,17 +1,26 @@
-import React, {Component} from "react";
+import * as React from "react";
 import {Grid, Row} from "react-bootstrap";
-import ClientNavbar from "./ClientNavbar.jsx";
 
-class ClientCore extends Component {
-  constructor(props) {
+import ClientNavbar from "./ClientNavbar";
+import IActiveKeyState from "./IActiveKeyState";
+
+interface IClientCoreComponents {
+  chat: JSX.Element;
+  setting: JSX.Element;
+}
+
+class ClientCore extends React.Component<any, IActiveKeyState> {
```

Comment:
Generic is there because that's how it's defined in the type definition. I think is React.Component<P, S>
IActiveKeyState is what I created to satisfy TypeScript while making it shared between two files. Open to modification or even deletion